counterfit/core/frameworks.py

```diff
@@ -35,18 +13,18 @@ def run(self):
         raise NotImplementedError
 
     @abstractmethod
-    def pre_attack_processing(self, cfattack: CFAttack):
+    def pre_attack_processing(self, cfattack):
         pass
 
     @abstractmethod
-    def post_attack_processing(self, cfattack: CFAttack):
+    def post_attack_processing(self, cfattack):
         pass
 
     @abstractmethod
-    def set_parameters(self, attack, parameters: dict) -> None:
-        raise NotImplementedError
+    def set_parameters(self, cfattack):
+        pass
```

Comment:
This function was marked as `abstractmethod` but not enforcing the functionality of @abstractmethod. We need to either enforce all sub-classes to implement this function and this function should `raise NotImplementedError ` or completely remove this function from this class and implement this function in respective subclasses.

Reply:
Great! Fixed in b545134f0b928c6db7cf935a785e0c2e3244cb6b.